### PR TITLE
Fix #21692 #21752 - logic in constructor of address validator and Locale Resolver check

### LINKED ISSUE #21692: Incorrect constructor of Magento\Sales\Model\Order\Address\Validator
### Summary (*)
Tested on 2.3, but exactly same logic is included in this constructor on 2.2-develop branch.
Constructor of `Magento\Sales\Model\Order\Address\Validator` includes logic and should not. The reason of it is that if this class is injected in constructor in Cli Command in some custom module, installation of Magento fails.

### Examples (*)

1. Add custom module with Cli Command in which `Magento\Sales\Model\Order\Address\Validator` is injected
2. Try to install Magento from scratch using Cli.
3. See following error: `Base table or view not found: 1146 Table 'database.eav_entity_type' doesn't exist, query was: SELECT `main_table`.* FROM `eav_entity_type` AS `main_table`"`

### Proposed solution
I'm preparing PR for this one.

### PR DESCRIPTION
**Pull request by Bartlomiejsz**
### Description (*)

This fixes issue #21692 - error is described there

### Fixed Issues (if relevant)
1. magento/magento2#21692: Incorrect constructor of Magento\Sales\Model\Order\Address\Validator
2. magento/magento2#21752: Error while installing Magento from scratch if Locale Resolver is injected in cli command

### Manual testing scenarios (*)

1. Add custom module with Cli Command and inject `Magento\Sales\Model\Order\Address\Validator` into its constructor
2. Run Magento installation from Cli
3. Verify if installation is finished with success

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @Bartlomiejsz. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-4482](https://jira.corp.magento.com/browse/ENGCOM-4482) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@Bartlomiejsz thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by soleksii** [comment]
Hi @Bartlomiejsz ! 

I've checked PR and it looks like changes does not solve the problem.

**Manual testing scenario:**
1. Add custom module with Cli Command and inject Magento\Sales\Model\Order\Address\Validator into its constructor;
2. Try to install Magento from scratch using Cli;
3. See following error:
problem

**Comment by Bartlomiejsz** [comment]
Hi @stoleksiy, this is another error. I wasn't aware that it also will appear for this class. I will add another issue and prepare PR later. Seems that those two are just combined and the other one have to be merged first.

**Comment by soleksii** [comment]
@Bartlomiejsz But, how can i check solution to the issue #21692 ? You need to solve the problem in scope of this PR.

**Comment by Bartlomiejsz** [comment]
@stoleksiy ok then, I will add fixing changes to this PR

**Comment by Bartlomiejsz** [comment]
@stoleksiy I've added fix for second issue

File: lib/internal/Magento/Framework/Locale/Resolver.php [hunk 1]
```diff
@@ -159,4 +169,18 @@ public function revert()
...
+
+    /**
+     * Retrieve Deployment Config
+     *
+     * @return DeploymentConfig
+     */
+    private function getDeploymentConfig()
+    {
+        if (!$this->deploymentConfig) {
+            $this->deploymentConfig = ObjectManager::getInstance()->get(DeploymentConfig::class);
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/Locale/Resolver.php]: Could you move deploymentConfig as last argument in constructor? It's not a good idea to get it through ObjectManager.

See "Adding a constructor parameter" section in https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/Locale/Resolver.php (hunk 1 above)
Comment (ihor-sviziev) [reply lib/internal/Magento/Framework/Locale/Resolver.php]: Also look at failing unit tests. Please fix them

Comment (Bartlomiejsz) [reply lib/internal/Magento/Framework/Locale/Resolver.php]: @ihor-sviziev please check again now

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-4482](https://jira.corp.magento.com/browse/ENGCOM-4482) has been created to process this Pull Request

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed 

**Result:**

after

**Pull request merged**